### DISCUSSION TITLE
Multiple XML sources?

### DISCUSSION THREAD

**Discussion by KryptoNyte-Sage** [OpenSageTV/Community#162] [Guide/EPG]
Hopefully NYP will see this.  Using Channels DVR (TVE sources) as a SageTV tuner has been working pretty well.

I'd like to get (2) different Channels DVR sources going in Sage, which appears to require (2) different XML configurations going, but I don't know where to start with this.
https://forums.sagetv.com/forums/showthread_t_66899_page_7.html?t=66899&page=7

**Comment by nyplayer** [comment]
IT is simple make of copy of the main XML file that you have working rename it to XMLTV.whateveryiuwant.properties ... replace the XML in the new one you created.
https://github.com/jzhvymetal/SageTv_XMLTVImportPlugin

**Comment by KryptoNyte-Sage** [comment]
So I replacing the original XML file with the new one?  Can I do this all in one XML file?

**Comment by nyplayer** [comment]
No you create a new XMLTV properties for each one for example XMLTV.Pluto.Properties , XMLTV.TVE.Properties etc ... Change the ID and Provider name.

Examples .. 

![image](https://github.com/user-attachments/assets/eb7c77a0-e6c0-45de-a920-20d7e84e40a5)

**Comment by KryptoNyte-Sage** [comment]
Ah.  So the first dot in your original example is omitted.

**Comment by KryptoNyte-Sage** [comment]
Edwin, I've tried just about every iteration of the xmltv filename, and the only thing that shows up in Sage for EPG (provider.name) when setting up a new tuner epg is from the original xmltv.properties file.

**Comment by nyplayer** [comment]
Can you post the internals of one you created ....

**Comment by KryptoNyte-Sage** [comment]
I found your post #222 on this page:
https://forums.sagetv.com/forums/showthread_t_66804_page_12.html?t=66804&page=12
and realized that the xmltv.properties file needs to be changed to point at the (2) other xmltv files that I created for each source (Philo and YoutubeTV via Channels DVR).  That seems to be working, but Sage seems to be having trouble dealing with duplicate channels in this scenario.  I may have to configure that manually in each tuner's available channels.

**Comment by KryptoNyte-Sage** [comment]
While I was testing just the Youtube TV source, a new problem popped up that I'll have to dig into.  When I added the first source YoutubeTV1 everything worked fine, but when I added the second YoutubeTV2 source, everything failed to tune.  This was not my experience previously when I added multiple sources for one service - so I dunno what's happening there.

**Comment by KryptoNyte-Sage** [comment]
I finally got (1) YoutubeTV source added and (1) Philo source added, but for some reason Sage is not resolving duplicate channels.  For duplicate channels, the guide is showing both of them.  I adjusted the priority putting Philo at 10 and Youtube at 1 and it doesn't resolve.  All duplicate channels are still showing.

Is there a setting somewhere in Sage that handles this, or will I need to adjust the lineups in Sage manually?

**Comment by KryptoNyte-Sage** [comment]
I did find this thread:
https://forums.sagetv.com/forums/showthread_t_14629.html?t=14629
That initially suggests duplicates from multiple xml sources will need to be handled manually, but then a user mentioned something near the end that I'll have to investigate.
